### ISSUE TITLE
Several issues with current column displays

### ISSUE THREAD

**Issue by shajoezhu** [insightsengineering/rlistings#11]
# Problem
## Data Setup

```{r}
library(dplyr)
library(scda)
library(rlistings)

admh <- synthetic_cdisc_data("rcd_2022_06_27")$admh

out <- admh %>%
  mutate(
    ASR = paste(AGE, SEX, RACE, sep = '/'),
    ID = paste(SITEID, SUBJID, sep = '/')
  ) %>%
  select(ID, ASR, TRT01A, MHSOC, MHTERM, TRTSDTM, ASTDTM, ASTDY, AENDTM, AENDY, ATIREL)

var_labels(out) <- c(
    ID = "Center/Patient ID", 
    ASR = "Age/Sex/Race", 
    TRT01A = "Treatment", 
    MHSOC = "SOC", 
    MHTERM = "Disease Term",
    TRTSDTM = "Date of First Study Drug Administration", 
    ASTDTM = "Start date of disease", 
    ASTDY = "Start day of disease", 
    AENDTM = "End date of disease", 
    AENDY = "End day of disease",  
    ATIREL = "Time relation of disease"
)
```

## Medical History and Concurrent Diseases

```{r}
lsting <- as_listing(
  out,
  key_cols = c("ID", "ASR", "TRT01A", "MHSOC", "MHTERM"),
  cols = names(out),
  main_title = "Medical History and Concurrent Diseases"
)

lsting
```

## output
```
> head(lsting, 20)
Medical History and Concurrent Diseases

————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————
Center/Patient ID            Age/Sex/Race              Treatment      SOC    Disease Term   Date of First Study Drug Administration   Start date of disease   Start day of disease   End date of disease   End day of disease   Time relation of disease
————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————
BRA-1/id-134        47/M/WHITE                       A: Drug X        cl A   trm A_2/2                2020-11-04 08:50:33                  2021-01-17                  74                2021-02-20               108              PRIOR_CONCOMITANT    
                                                                                                      2020-11-04 08:50:33                  2021-02-02                  90                2021-02-12               100              PRIOR_CONCOMITANT    
                                                                                                      2020-11-04 08:50:33                  2020-12-11                  37                2021-01-20                77              PRIOR_CONCOMITANT    
                                                                      cl D   trm D_2/3                2020-11-04 08:50:33                  2020-12-19                  45                2021-01-26                83              PRIOR_CONCOMITANT    
BRA-1/id-141        35/F/WHITE                       C: Combination                                   2020-07-25 19:10:56                  2021-11-30                 493                2021-12-23               516              PRIOR_CONCOMITANT    
                                                                      cl A   trm A_1/2                2020-07-25 19:10:56                  2022-04-21                 635                2022-06-03               678              PRIOR_CONCOMITANT    
                                                                      cl A                            2020-07-25 19:10:56                  2022-03-17                 600                2022-06-18               693              PRIOR_CONCOMITANT    
                                                     C: Combination   cl B   trm B_2/3                2020-07-25 19:10:56                  2021-04-10                 259                2021-04-17               266              PRIOR_CONCOMITANT    
                                                                             trm D_1/3                2020-07-25 19:10:56                  2022-07-25                 730                2022-07-26               731              PRIOR_CONCOMITANT    
                                                                             trm D_3/3                2020-07-25 19:10:56                  2021-05-24                 303                2022-07-18               723              PRIOR_CONCOMITANT    
BRA-1/id-236        32/M/BLACK OR AFRICAN AMERICAN                    cl B                            2021-01-15 13:37:05                  2021-02-06                  22                2022-03-23               432              PRIOR_CONCOMITANT    
                                                                                                      2021-01-15 13:37:05                  2021-07-15                 181                2022-12-14               698              PRIOR_CONCOMITANT    
                                                                      cl B   trm B_1/3                2021-01-15 13:37:05                  2022-12-18                 702                2022-12-22               706              PRIOR_CONCOMITANT    
BRA-1/id-265
```

**Comment by shajoezhu** [comment]
Issue: missing SOC items, 

For patient `BRA-1/id-134`, `SOC` should include three items, currently only "cl A" and "cl D" printed

```
> out %>% filter(ID == "BRA-1/id-134")
# A tibble: 4 × 11
  ID           ASR        TRT01A    MHSOC MHTERM    TRTSDTM             ASTDTM              ASTDY AENDTM              AENDY ATIREL           
  <chr>        <chr>      <fct>     <fct> <fct>     <dttm>              <dttm>              <int> <dttm>              <int> <fct>            
1 BRA-1/id-134 47/M/WHITE A: Drug X cl B  trm B_2/3 2020-11-04 08:50:33 2020-12-11 00:00:00    37 2021-01-20 00:00:00    77 PRIOR_CONCOMITANT
2 BRA-1/id-134 47/M/WHITE A: Drug X cl D  trm D_2/3 2020-11-04 08:50:33 2020-12-19 00:00:00    45 2021-01-26 00:00:00    83 PRIOR_CONCOMITANT
3 BRA-1/id-134 47/M/WHITE A: Drug X cl A  trm A_2/2 2020-11-04 08:50:33 2021-01-17 00:00:00    74 2021-02-20 00:00:00   108 PRIOR_CONCOMITANT
4 BRA-1/id-134 47/M/WHITE A: Drug X cl A  trm A_2/2 2020-11-04 08:50:33 2021-02-02 00:00:00    90 2021-02-12 00:00:00   100 PRIOR_CONCOMITANT
```

**Comment by shajoezhu** [comment]
Issue: repeated treatment display for the same patient

For patient `BRA-1/id-141`, it only has "C: Combination", but it is printed twice.

**Comment by shajoezhu** [comment]
Issue: Missing treatment

For patient `BRA-1/id-236`, "B: Placebo" arm is not displayed

**Comment by shajoezhu** [comment]
Issue: Rendering taking forever

just call `lsting` from the console takes forever to render, need to trim it with `head(lsting, 20)`

**Comment by gmbecker** [comment]
These issues are all now closed, though with the caveat of #16, which I consider a separate issue.
